src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/OffenderSearchService.kt

```diff
@@ -29,9 +30,20 @@ class OffenderSearchService(private val offenderSearchWebClient: WebClient) {
         )
         .retrieve()
         .awaitBody<OffenderSearchPrisonerList>()
-      offenders.addAll(pageOfData.content)
+      emit(pageOfData.content)
       page += 1
     } while (!pageOfData.last)
+  }
+
+  /**
+   * Searches for offenders using a cell location prefix (e.g. MDI-1) returning a complete list
+   * Requires role PRISONER_IN_PRISON_SEARCH or PRISONER_SEARCH
+   */
+  suspend fun getOffendersAtLocation(prisonId: String, cellLocationPrefix: String): List<OffenderSearchPrisoner> {
+    val offenders = mutableListOf<OffenderSearchPrisoner>()
+    findOffendersAtLocation(prisonId, cellLocationPrefix).collect {
+      offenders.addAll(it)
+    }
```

Comment:
`.toList().flatten()` should do the same thing without having to use a mutable list (may be a bit more performant too?) - but check. There may be an even nicer way of doing this too (there are a bunch of `flat*` method on `Flow` but this seems clear enough)

Reply:
>may be a bit more performant too?
i would expect the opposite! my version only creates 1 list and appends to it; to-list + flatten would first make a nested list and then do the same thing my version does

Reply:
i looked around for a flatten-into-something type methods but didn't find any :(

Reply:
I guess my point was, I'd imagine library methods/potentially the compiler to be clever enough to "look" though these and be able to avoid unnecessary allocations/concatenations (and it reads better).

There was something like `flatMapMerge` but we don't need to transform it and I didn't quite got how it works tbh.